File: PDOO/PDOO/Java/Civitas/src/civitas/Dado.java

package civitas;

import java.util.ArrayList;
import java.util.Random;


public class Dado {
    
    private Random random;
    private int ultimoResultado;
    private Boolean debug;
    
    static final private Dado instance = new Dado();
    private static int SalidaCarcel = 5;
    
    private Dado(){
        debug = true;
        ultimoResultado = 0;
        random = new Random();
     }
    
    static public Dado getInstance(){
        return instance;
    }
    
    int tirar(){
        int resultado;
        if (debug == false){
            resultado = random.nextInt(6)+1; //el +1 es para que el dado vaya de 1 a 6 (es decir, no tome el valor 0)
        } else {
            resultado = 1;
        }
        ultimoResultado = resultado;
        return resultado;
    }
    
    Boolean salgoDeLaCarcel(){
        Boolean resultado = false;
        if (getUltimoResultado() == 6){
            resultado = true;
        }
        return resultado;
    }
    
    int quienEmpieza(int n){
        int resultado;
        resultado = random.nextInt(n);
        return resultado;
    }
    
    /*pacage*/ public void setDebug(Boolean d){
        debug = d;
        if (debug == true){
            Diario.getInstance().ocurreEvento("dado debug true");
        } else {
            Diario.getInstance().ocurreEvento("dado debug false");
        }
    }
    
    int getUltimoResultado(){
        return ultimoResultado;
    }
    
}

File: PDOO/p4/civitas_rb/lib/dado.rb
require_relative 'diario'
require 'singleton'

module Civitas
class Dado
  include Singleton
  
  def initialize
    @debug=true
    @ultimoResultado=0
  end
  
  Dado.instance
  
  def tirar
    if @debug==true
      resultado=1
    else
      resultado= rand(0..5) + 1
    end
    @ultimoResultado=resultado
    return resultado
  end
  
  def salgo_de_la_carcel()
    resultado=false
    if tirar()==6
      resultado=true
    end
    return resultado
  end
  
  def quien_empieza(n)
    return rand(0...n)
  end
  
  def setdebug(d)
    @debug=d
    if @debug==true
      Diario.instance.ocurre_evento("dado debug true")
    else
      Diario.instance.ocurre_evento("dado debug false")
    end
  end
  
  attr_accessor :debug, :SalirCarcel, :ultimoResultado
  
end
end